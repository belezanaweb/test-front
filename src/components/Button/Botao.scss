@import '../../assets/sass/colors';

.botao {
  width: 100%;
  font-weight: 700;
  font-size: 20px;
  letter-spacing: 0.5px;
  height: 60px;
  font-size: 20px;
  font-weight: 700;
  color: #fff;
  background: $primaryColor;
  display: block;
  text-transform: uppercase;
  border: 0;
  box-shadow: 0 2px 4px rgba(0, 0, 0, .25), #D45A00 0 -3px 0 inset;
  border-radius: 3px;
}
