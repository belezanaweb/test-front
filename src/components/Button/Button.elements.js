import styled from 'styled-components'

export const ButtonStyled = styled.button`
  cursor: pointer;
  border: none;
  outline: none;
  font-size: 1.25rem;
  line-height: 1.5rem;
  letter-spacing: 0.5px;
  text-align: center;
  width: 100%;
  height: 3.75rem;
  color: #fff;
  border-radius: 3px;
  background-color: #ff6c00;
  box-shadow: 0 3px 0 #d45a00, 0 2px 4px rgba(0, 0, 0, 0.25);

  &:hover {
    background-color: #d45a00;
  }
`
