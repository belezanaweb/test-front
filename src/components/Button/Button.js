import React from 'react'

const Button = () => {
  return (
    <div>
      <button></button>
    </div>
  )
}

export default Button
