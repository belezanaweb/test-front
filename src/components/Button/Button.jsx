import React from 'react'

import { StyledButton } from './styles'

function Button({ children, ...rest }) {
  return <StyledButton {...rest}>{children}</StyledButton>
}

export default Button
