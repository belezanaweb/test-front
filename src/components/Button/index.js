import React from 'react'
import { StyledButton } from './styles'

const Button = ({ title, onClick, ...props }) => {
  return (
    <StyledButton {...props} onClick={onClick}>
      {title}
    </StyledButton>
  )
}

export default Button
