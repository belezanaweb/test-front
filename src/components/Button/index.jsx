import React from 'react'
import * as S from './styled'
import { useNavigate } from 'react-router-dom'

const Button = ({ text, path, disabled }) => {
  let navigate = useNavigate()
  return (
    <S.Button onClick={() => navigate(`${path}`)} disabled={disabled}>
      {text}
    </S.Button>
  )
}

export default Button
