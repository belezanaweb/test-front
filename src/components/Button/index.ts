import styled from 'styled-components'

export const Button = styled.button`
  width: 100%;
  background-color: ${(props) => props.theme.light.colors['--c-contrast-two']};
  border: none;
  padding: 15px 0;
  border-radius: 3px;

  box-shadow: inset 0 -3px 0 0 ${(props) => props.theme.light.colors['--c-contrast-tree']},
    1px 1px 10px 1px rgb(0 0 29 / 12%);

  color: white;
  font-size: ${(props) => props.theme.fontSizes.extraLarge};
  font-weight: bold;
  text-transform: uppercase;
  text-decoration: none;
  text-align: center;

  &:hover {
    background-color: ${(props) => props.theme.light.colors['--c-contrast-tree']};
  }

  &:disabled {
    background-color: ${(props) => props.theme.light.colors['--c-contrast-tree']};
  }
`
