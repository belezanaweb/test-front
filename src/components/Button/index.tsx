import { ButtonProps } from './interfaces'
import * as Styled from './styled'

const Button = ({ onClick, text, fullWidth }: ButtonProps) => {
  return (
    <Styled.Button onClick={onClick} fullWidth={fullWidth}>
      {text}
    </Styled.Button>
  )
}

export default Button
