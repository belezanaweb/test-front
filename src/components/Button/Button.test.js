import React from 'react'
import { render } from '@testing-library/react'
import Button from '.'

describe('Button', () => {
  it('should render [snapshot]', () => {
    const { container } = render(<Button />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with disabled attribute [snapshot]', () => {
    const { container } = render(<Button disabled />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
