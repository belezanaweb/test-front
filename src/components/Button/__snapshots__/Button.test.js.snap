// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render [snapshot] 1`] = `
.emotion-0 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  background-color: #ff6c00;
  border: 0;
  border-radius: 3px;
  box-shadow: inset 0 -3px 0 0 #d45a00,0 2px 4px 0 rgba(0,0,0,0.25);
  color: #fff;
  cursor: pointer;
  display: block;
  font-size: 20px;
  font-weight: bold;
  line-height: 24px;
  padding: 18px 24px;
  -webkit-transition: all 0.5s linear;
  transition: all 0.5s linear;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
}

.emotion-0.hover,
.emotion-0:hover,
.emotion-0:active {
  background-color: #d45a00;
  box-shadow: inset 0 0 0 0 #d45a00,0 0 0 0 rgba(0,0,0,0.25);
}

.emotion-0:disabled {
  background-color: #888;
  box-shadow: inset 0 -3px 0 0 #666,0 2px 4px 0 rgba(0,0,0,0.25);
  color: #ccc;
  cursor: not-allowed;
}

<button
  class="emotion-0 emotion-1"
/>
`;

exports[`Button should render with disabled attribute [snapshot] 1`] = `
.emotion-0 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  background-color: #ff6c00;
  border: 0;
  border-radius: 3px;
  box-shadow: inset 0 -3px 0 0 #d45a00,0 2px 4px 0 rgba(0,0,0,0.25);
  color: #fff;
  cursor: pointer;
  display: block;
  font-size: 20px;
  font-weight: bold;
  line-height: 24px;
  padding: 18px 24px;
  -webkit-transition: all 0.5s linear;
  transition: all 0.5s linear;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
}

.emotion-0.hover,
.emotion-0:hover,
.emotion-0:active {
  background-color: #d45a00;
  box-shadow: inset 0 0 0 0 #d45a00,0 0 0 0 rgba(0,0,0,0.25);
}

.emotion-0:disabled {
  background-color: #888;
  box-shadow: inset 0 -3px 0 0 #666,0 2px 4px 0 rgba(0,0,0,0.25);
  color: #ccc;
  cursor: not-allowed;
}

<button
  class="emotion-0 emotion-1"
  disabled=""
/>
`;
